main.go

```diff
@@ -101,37 +107,47 @@ func main() {
 		// loop indefinitely
 		for {
 			log.Info().Msg("Watching services for all namespaces...")
-			watcher, err := client.CoreV1().WatchServices(context.Background(), k8s.AllNamespaces)
+			timeoutSeconds := int64(300)
+			watcher, err := kubeClientset.CoreV1().Services("").Watch(metav1.ListOptions{
+				TimeoutSeconds: &timeoutSeconds,
+			})
 			if err != nil {
 				log.Error().Err(err).Msg("WatchServices call failed")
 			} else {
 				// loop indefinitely, unless it errors
 				for {
-					event, service, err := watcher.Next()
-					if err != nil {
-						log.Error().Err(err).Msg("Getting next event from service watcher failed")
+					event, ok := <-watcher.ResultChan()
+					if !ok {
+						log.Warn().Msg("Watcher for services is closed")
 						break
 					}
 
-					if *event.Type == k8s.EventAdded || *event.Type == k8s.EventModified || *event.Type == k8s.EventDeleted {
-						if *event.Type == k8s.EventDeleted {
+					if event.Type == watch.Added || event.Type == watch.Modified || event.Type == watch.Deleted {
+
+						service, ok := event.Object.(*v1.Service)
+						if !ok {
+							log.Warn().Msg("Watcher for pods returns event object of incorrect type")
```

Comment:
typo here? Shouldn't it be "Watcher for services..." ?

Reply:
Heh heh, serves me for copying this from elsewhere...